### PR TITLE
Add classes for redaction support

### PR DESCRIPTION
**Pull request by evanfuller**

### CODE REVIEW

File: src/main/java/ai/nightfall/scan/model/redaction/CryptoConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public CryptoConfig(String publicKey) {
+        this.publicKey = publicKey;
+    }
+
+    /**
+     * Get the public key.
+     *
+     * @return the public key
+     */
+    public String getPublicKey() {
```
Comment (galennightfall) [inline src/main/java/ai/nightfall/scan/model/redaction/CryptoConfig.java:28]: would it make sense to normalize keys? Maybe not.

**Review by galennightfall — COMMENTED** [review]
So much boilerplate. How distasteful. Looks fine.
I don't see `RedactionConfig` used apart from as a field. Does it just get magically serialized with the `@JsonProperty` magic?

File: src/main/java/ai/nightfall/scan/model/DetectionRule.java [hunk 2]
```diff
@@ -21,6 +21,30 @@
     @JsonProperty("name")
     private String name;
 
+    /**
+     * Create a detection rule with the provided detectors and logicalOp.
+     *
+     * @param detectors a list of detectors
+     * @param logicalOp a logical op: ANY or ALL
+     */
+    public DetectionRule(List<Detector> detectors, String logicalOp) {
```
[the remark this answers is not in the document]
Comment (evanfuller) [reply src/main/java/ai/nightfall/scan/model/DetectionRule.java:30]: yes, good call. i'll do this for confidence as well.

**Comment by evanfuller** [comment]
> So much boilerplate. How distasteful. Looks fine. I don't see `RedactionConfig` used apart from as a field. Does it just get magically serialized with the `@JsonProperty` magic?

@galennightfall Yes -- this is just a field that can be configured in a request object that a customer provides.

I'm planning on uploading some examples of using the SDK to this repo in a follow-up PR.

**Review by galennightfall — APPROVED** [review]
:shipit:

**Pull request merged**